src/handlers/RF24StreamLogHandler.h

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file RF24StreamLogHandler.h
+ *
+ * Created on: 2 Oct 2020
+ *     Author: Witold Markowski (wmarkow)
+ *
+ * Copyright (C)
+ *    2020        Witold Markowski (wmarkow)
+ *
+ * This General Public License does not permit incorporating your program into
+ * proprietary programs.  If your program is a subroutine library, you may
+ * consider it more useful to permit linking proprietary applications with the
+ * library.  If this is what you want to do, use the GNU Lesser General
+ * Public License instead of this License.
+ */
+
+#ifndef SRC_HANDLERS_RF24STREAMLOGHANDLER_H_
+#define SRC_HANDLERS_RF24STREAMLOGHANDLER_H_
+
+#include <RF24LogHandler.h>
+#include <Stream.h>
+
+/**
+ * A log handler implementation which outputs log messages to a stream.
+ */
+class RF24StreamLogHandler : public RF24LogHandler
+{
+private:
+   Stream *stream;
+   uint8_t logLevel;
+   void appendTimestamp();
+   void appendLogLevel(uint8_t logLevel);
+   void appendVendorId(const __FlashStringHelper *vendorId);
+
+public:
+   RF24StreamLogHandler(Stream *stream);
+   void log(uint8_t logLevel, const __FlashStringHelper *vendorId,
+         const char *message, va_list *args);
+
+   void log(uint8_t logLevel, const __FlashStringHelper *vendorId,
+         const __FlashStringHelper *message, va_list *args);
+
+   void setLogLevel(uint8_t logLevel);

```

Comment:
Yes, I just see ```RF24LogHandler``` to be just a pure abstraction - a set of public methods. Since this class is directly used in ```RF24Logger``` for logging purposes. It is a duty of the end user application to provide an implementation of ```RF24LogHandler```.

In Java world there exists a ```interface``` keyword for this, so you would define this as:
```
interface RF24LogHandler
{
    void setLogLevel(uint8_t logLevel);
}
```

How is this done in C++? With abstract class and virtual methods?

Answering your question: maybe we could move the ```setLoglevel(uint8_8)```  and also ```uint8_t logLevel;``` variable directly to **RF24LogHandler.h**. Then we need to provide ```RF24LogHandler.cpp``` with a default implementation (just for storing the variable), but anyway the final handler needs to take care and check the log level and make a decision if the log should be logged or not.

I do not know. Need to think about it.

Reply:
>How is this done in C++? With abstract class and virtual methods?

its been a long time since a i looked at javadocs, but a quick search compared `interface` to inheritance. Additionally an abstract class can still have virtually defined methods. If only 1 method is pure virtual (using the `retType func() = 0;`), then the whole class is still considered an abstract class. In the end if `setLevel()` is going to do the same thing every time, it might be better to define it in the abstract class. 

The only scenario I can think of when the end user would want to override `setLevel()` is if they want to clamp/limit the log levels to a specific range (instead of using the whole byte that's allocated anyway).

Reply:
For now I made all methods of ```RF24Handler``` a **pure virtual** (using the ```retType func() = 0;```).